# Exploratory Data Analysis & Statistical Insights

RabTech Academy - Data Analytics and business intelligence

## 1. Introduction

Exploratory Data Analysis (EDA) is the process of examining, summarizing, and visualizing a dataset to understand its underlying structure, distribution, relationships, and unusual observations.

In this analysis, the cleaned payment transaction dataset is investigated using descriptive statistics, statistical visualizations, correlation analysis, and hypothesis testing.

The main objectives of this analysis are to:

1. Summarize the numerical characteristics of the dataset using descriptive statistics.
2. Examine distributions and identify potential anomalies using histograms and box plots.
3. Investigate relationships between numerical financial variables using correlation analysis.
4. Formulate and statistically test three business-related hypotheses.
5. Summarize the most important findings and translate statistical results into meaningful business insights.

The analysis follows a structured workflow:

**Data → Descriptive Statistics → Visualization → Statistical Testing → Business Interpretation**

In [1]:
# Import required libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats
from IPython.display import display

### Purpose of the Libraries

The following Python libraries are used throughout the analysis:

- **Pandas:** Data loading, manipulation, filtering, and statistical summarization.
- **NumPy:** Numerical operations and support for numerical data processing.
- **Matplotlib:** General-purpose data visualization.
- **Seaborn:** Statistical visualization, including histograms, box plots, and correlation heatmaps.
- **SciPy:** Statistical hypothesis testing.
- **IPython.display:** Used to display Pandas tables clearly inside the notebook.

## 2. Load the Clean Dataset

The analysis is performed on the cleaned dataset generated during the previous data-cleaning stage.

Using the cleaned dataset ensures that the EDA and statistical analysis are based on data that has already undergone the required data-quality procedures, such as handling missing and duplicate primary keys and applying the agreed cleaning logic.

The original raw dataset is therefore not used directly for this stage.

In [2]:
# Load the cleaned dataset

df = pd.read_csv(
    r"E:\CODING\Py\submission_02_reb_tech_academy\clean_dataset.csv"
)

print("Clean dataset loaded successfully.")

Clean dataset loaded successfully.


## 3. Initial Dataset Inspection

Before performing statistical analysis, the structure and dimensions of the cleaned dataset are inspected.

This initial inspection helps establish:

- the number of observations,
- the number of variables,
- variable data types,
- the presence of missing values,
- and the general structure of the dataset.

Understanding the dataset structure is important because statistical methods should be applied according to the type and meaning of each variable.

In [3]:
# Display dataset dimensions

print(f"Number of observations: {df.shape[0]}")
print(f"Number of variables: {df.shape[1]}")

Number of observations: 90000
Number of variables: 15


In [4]:
# Inspect variable names and data types

df.info()

<class 'pandas.DataFrame'>
RangeIndex: 90000 entries, 0 to 89999
Data columns (total 15 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   payment_id             90000 non-null  str    
 1   order_id               90000 non-null  str    
 2   payment_date           90000 non-null  str    
 3   payment_method         90000 non-null  str    
 4   payment_status         90000 non-null  str    
 5   amount_paid            90000 non-null  float64
 6   transaction_fee        90000 non-null  float64
 7   refund_amount          90000 non-null  float64
 8   order_month            90000 non-null  int64  
 9   order_month_name       90000 non-null  str    
 10  order_year             90000 non-null  int64  
 11  quarter                90000 non-null  str    
 12  net_settlement_amount  90000 non-null  float64
 13  retention_margin_pct   90000 non-null  float64
 14  profitability_status   90000 non-null  str    
dtypes: float64(5)

In [5]:
# Display the first five observations

display(df.head())

,payment_id,order_id,payment_date,payment_method,payment_status,amount_paid,transaction_fee,refund_amount,order_month,order_month_name,order_year,quarter,net_settlement_amount,retention_margin_pct,profitability_status
0,PAY1000000,ORD1000000,2023-02-12,Emi,Failed,0.00,0.0,0.0,2,February,2023,2023Q1,0.00,0.0,Break-Even
1,PAY1000001,ORD1000001,2023-02-12,Upi,Success,4998.00,0.0,0.0,2,February,2023,2023Q1,4998.00,100.0,Profitable
2,PAY1000002,ORD1000002,2024-12-14,Net Banking,Success,867.30,0.0,0.0,12,December,2024,2024Q4,867.30,100.0,Profitable
3,PAY1000003,ORD1000003,2024-12-14,Cash On Delivery,Not Charged,0.00,0.0,0.0,12,December,2024,2024Q4,0.00,0.0,Break-Even
4,PAY1000005,ORD1000005,2025-01-08,Cash On Delivery,Success,8275.89,0.0,0.0,1,January,2025,2025Q1,8275.89,100.0,Profitable


In [6]:
# Check missing values

missing_values = df.isnull().sum()

display(
    missing_values[missing_values > 0].to_frame("Missing Values")
)

,Missing Values


## 4. Summary Descriptive Statistics

### 4.1 Selection of Numerical Variables

Descriptive statistics such as mean, median, standard deviation, and quartiles are primarily used to summarize quantitative variables.

For this analysis, five financially and operationally meaningful numerical variables are selected:

- **Amount Paid:** The amount paid for a transaction.
- **Transaction Fee:** The fee associated with the transaction.
- **Refund Amount:** The monetary amount refunded to the customer.
- **Net Settlement Amount:** The amount remaining after relevant transaction adjustments.
- **Retention Margin (%):** The retention-related percentage measure available in the dataset.

Identifier variables and categorical variables are excluded from the descriptive-statistics analysis because calculating measures such as the mean of an identifier or payment method code would not provide meaningful business information.

This distinction is important because a variable being stored as a number does not automatically make it a meaningful quantitative measurement.

In [7]:
# Define meaningful numerical variables for descriptive analysis

num_cols = [
    'amount_paid',
    'transaction_fee',
    'refund_amount',
    'net_settlement_amount',
    'retention_margin_pct'
]

print("Numerical variables selected for analysis:")
for col in num_cols:
    print(f"- {col}")

Numerical variables selected for analysis:
- amount_paid
- transaction_fee
- refund_amount
- net_settlement_amount
- retention_margin_pct


### Analytical Relevance of Selected Variables

The selected variables represent different financial dimensions of the payment process.

| Variable | Analytical purpose |
|---|---|
| Amount Paid | Measures transaction value |
| Transaction Fee | Measures transaction-related cost |
| Refund Amount | Measures returned monetary value |
| Net Settlement Amount | Measures the resulting settlement value |
| Retention Margin (%) | Measures the percentage-based retention characteristic |

Together, these variables provide a numerical foundation for understanding the scale, variability, and distribution of the financial transactions in the dataset.

### 4.2 Descriptive Statistics

Descriptive statistics provide a compact numerical summary of the selected variables.

The analysis considers:

- **Mean:** arithmetic average of the observations.
- **Median:** middle value of the ordered observations.
- **Standard deviation:** measures the dispersion of observations around the mean.
- **Quartiles:** identify the 25th, 50th, and 75th percentiles.
- **Minimum and maximum:** indicate the observed range.
- **Skewness:** provides an indication of asymmetry in the distribution.

The median and quartiles are particularly useful when a variable contains extreme observations because they are less influenced by outliers than the mean.

In [8]:
# Calculate descriptive statistics

desc_stats = df[num_cols].describe().T

# Add median and skewness
desc_stats['median'] = df[num_cols].median()
desc_stats['skewness'] = df[num_cols].skew()

# Display the results

display(
    desc_stats[
        [
            'count',
            'mean',
            'std',
            'min',
            '25%',
            'median',
            '75%',
            'max',
            'skewness'
        ]
    ]
)

,count,mean,std,min,25%,median,75%,max,skewness
amount_paid,90000.0,22074.812571,41625.655600,0.000000,3271.355000,8275.890,19958.4000,607058.55,4.163360
transaction_fee,90000.0,73.758924,320.075004,0.000000,0.000000,0.000,0.0000,7936.84,8.444339
refund_amount,90000.0,2156.643783,14640.937742,0.000000,0.000000,0.000,0.0000,541686.50,12.771816
net_settlement_amount,90000.0,19844.409864,40484.776585,-317282.330000,2000.067500,8163.545,17559.7525,607058.55,4.157830
retention_margin_pct,90000.0,82.756673,68.199396,-3833.815215,98.500009,100.000,100.0000,100.00,-23.536856


### 4.3 Interquartile Range

The Interquartile Range (IQR) measures the spread of the middle 50% of observations.

It is calculated as:

\[
IQR = Q_3 - Q_1
\]

where:

- \(Q_1\) is the 25th percentile.
- \(Q_3\) is the 75th percentile.

The IQR is less sensitive to extreme observations than the range and is therefore useful when investigating potential outliers.

In [9]:
# Calculate quartiles and interquartile range (IQR)

q1 = df[num_cols].quantile(0.25)
q3 = df[num_cols].quantile(0.75)

iqr = q3 - q1

quartile_summary = pd.DataFrame({
    'Q1 (25%)': q1,
    'Median (Q2)': df[num_cols].median(),
    'Q3 (75%)': q3,
    'IQR': iqr
})

display(quartile_summary)

,Q1 (25%),Median (Q2),Q3 (75%),IQR
amount_paid,3271.355000,8275.890,19958.4000,16687.045000
transaction_fee,0.000000,0.000,0.0000,0.000000
refund_amount,0.000000,0.000,0.0000,0.000000
net_settlement_amount,2000.067500,8163.545,17559.7525,15559.685000
retention_margin_pct,98.500009,100.000,100.0000,1.499991


### 4.4 Interpretation of Descriptive Statistics

The descriptive statistics provide an initial understanding of the financial characteristics of the dataset.

The interpretation focuses on four key areas:

1. **Central tendency:** Mean and median are compared to understand the typical transaction level.
2. **Variability:** Standard deviation is examined to determine how widely observations vary around the mean.
3. **Distribution:** Q1, median, and Q3 are used to understand where the central 50% of observations are located.
4. **Potential skewness:** The relationship between mean, median, and skewness is examined to identify possible asymmetry.

Particular attention is given to cases where the mean is substantially different from the median, as this may indicate the influence of unusually large or small observations.

The IQR is also considered because it describes the spread of the middle 50% of observations and provides a robust basis for subsequent outlier analysis.

In [10]:
desc_stats

,count,mean,std,min,25%,50%,75%,max,median,skewness
amount_paid,90000.0,22074.812571,41625.655600,0.000000,3271.355000,8275.890,19958.4000,607058.55,8275.890,4.163360
transaction_fee,90000.0,73.758924,320.075004,0.000000,0.000000,0.000,0.0000,7936.84,0.000,8.444339
refund_amount,90000.0,2156.643783,14640.937742,0.000000,0.000000,0.000,0.0000,541686.50,0.000,12.771816
net_settlement_amount,90000.0,19844.409864,40484.776585,-317282.330000,2000.067500,8163.545,17559.7525,607058.55,8163.545,4.157830
retention_margin_pct,90000.0,82.756673,68.199396,-3833.815215,98.500009,100.000,100.0000,100.00,100.000,-23.536856


In [11]:
quartile_summary

,Q1 (25%),Median (Q2),Q3 (75%),IQR
amount_paid,3271.355000,8275.890,19958.4000,16687.045000
transaction_fee,0.000000,0.000,0.0000,0.000000
refund_amount,0.000000,0.000,0.0000,0.000000
net_settlement_amount,2000.067500,8163.545,17559.7525,15559.685000
retention_margin_pct,98.500009,100.000,100.0000,1.499991


### 4.4 Interpretation of Descriptive Statistics

The descriptive statistics reveal substantial variation and non-normality across the selected financial variables.

**Amount Paid:**  
The mean amount paid is ₹22,074.81, while the median is considerably lower at ₹8,275.89. The standard deviation of ₹41,625.66 and positive skewness of 4.16 indicate substantial variability and a strongly right-skewed distribution. The maximum transaction value of ₹607,058.55 is considerably higher than the median, suggesting that a relatively small number of high-value transactions influence the mean.

**Transaction Fee:**  
Transaction fees have a mean of ₹73.76 but a median of ₹0. The first and third quartiles are also both zero, indicating that at least 75% of observations have zero transaction fees. The high positive skewness of 8.44 suggests that fees are concentrated at zero with a smaller number of transactions generating substantially higher fees.

**Refund Amount:**  
Refund amounts show a similar but more extreme pattern. The mean is ₹2,156.64, whereas the median and both Q1 and Q3 are zero. This indicates that most observations have no refund, while a relatively small number of transactions contain large refund amounts. The maximum refund of ₹541,686.50 and skewness of 12.77 demonstrate strong right-skewness.

**Net Settlement Amount:**  
The net settlement amount has a mean of ₹19,844.41 and a median of ₹8,163.55. Its positive skewness of 4.16 and high standard deviation of ₹40,484.78 indicate considerable variability and right-skewness. The minimum value of -₹317,282.33 is unusual and requires further investigation to determine whether it represents legitimate settlement adjustments or a potential data-quality or calculation issue.

**Retention Margin (%):**  
Retention margin shows a substantially different distribution. The median and third quartile are both 100%, while the mean is lower at 82.76%. The minimum value of -3,833.82% and strongly negative skewness of -23.54 indicate the presence of extreme low-value observations that pull the mean downward. These observations require further investigation to determine whether they are legitimate outcomes of the underlying calculation or potential anomalies.

Overall, the descriptive statistics indicate that the financial variables are highly skewed rather than normally distributed. Mean values should therefore be interpreted alongside medians, quartiles, and standard deviations. The concentration of zero values in transaction fees and refunds, the extreme values in financial variables, and the unusual retention-margin observations provide important areas for further investigation in the visualization stage.